realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -406,6 +451,10 @@ public DynamicRealmObject getObject(String fieldName) {
             return RealmFieldType.FLOAT_LIST;
         } else if (primitiveType.equals(Double.class)) {
             return RealmFieldType.DOUBLE_LIST;
+        } else if (primitiveType.equals(Decimal128.class)) {
```

Comment:
As discussed, let's add the bson dependency for this PR and optimise later a mechanism to make it optional. I think the method count and size will not affect apps not using these types (because of R8 tree shaking)